variables.tf

```diff
@@ -68,22 +68,64 @@ variable "s3_folder" {
   description = "The Amazon S3 bucket folder to use as path of the exported data"
 }
 
-variable "s3_lifecycle_configuration_rules" {
+variable "s3_lifecycle_rules" {
   type = list(object({
-    enabled = bool
-    id      = string
-
-    abort_incomplete_multipart_upload_days = number
-
-    filter_and = any
-    expiration = any
-    transition = list(any)
-
-    noncurrent_version_expiration = any
-    noncurrent_version_transition = list(any)
+    enabled                                = optional(bool, true)
+    id                                     = optional(string)
+    abort_incomplete_multipart_upload_days = optional(number)
+
+    filter_and = optional(object({
+      object_size_greater_than = optional(number)
+      object_size_less_than    = optional(number)
+      prefix                   = optional(string)
+      tags                     = optional(map(string), {})
+    }))
+
+    expiration = optional(object({
+      date = optional(string)
+      days = optional(number)
+    }))
+
+    expired_object_delete_marker = optional(bool)
```

Comment:
looks like variable type is wrongly defined, mb just make copy-paste of variable from the [origin s3 module](https://github.com/cloudposse/terraform-aws-s3-bucket/blob/f08e5120219dc877202b6a247c646d19503c9d04/variables.tf#L142) , what do you think? 

```
       rule {
           id     = "rds-s3-export-rotation"
           status = "Enabled"

           expiration {
               days                         = 180
               expired_object_delete_marker = (known after apply)
            }

           filter {
            }

           transition {
               days          = 60
               storage_class = "GLACIER"
            }
        }
       rule {
           id     = "rds-s3-export-delete-expiration-markers"
           status = "Enabled"

           abort_incomplete_multipart_upload {
               days_after_initiation = 3
            }

           filter {
            }
        }
```